nalgebra-sparse/src/io/matrix_market.rs

```diff
@@ -1329,3 +1417,110 @@ fn next_dense_coordinate(
         }
     }
 }
+
+/// Write a sparse matrix into Matrix Market format string.
+///
+/// Our exporter only writes matrix into `coordiante` and `general` format.
+///
+///
+/// Examples
+/// --------
+/// ```
+/// # use matrixcompare::assert_matrix_eq;
+/// use nalgebra_sparse::io::{write_to_matrix_market_str,load_coo_from_matrix_market_str};
+/// let str = r#"
+/// %%matrixmarket matrix coordinate integer general
+/// 5 4 2
+/// 1 1 10
+/// 2 3 5
+/// "#;
+/// let matrix = load_coo_from_matrix_market_str::<i32>(&str).unwrap();
+/// let generated_matrixmarket_string = write_to_matrix_market_str(&matrix);
+/// // 'generated_matrixmarket' should equal to the 'matrix'
+/// let generated_matrixmarket = load_coo_from_matrix_market_str::<i32>(&generated_matrixmarket_string).unwrap();
+/// assert_matrix_eq!(matrix,generated_matrixmarket);
+/// ```
+pub fn write_to_matrix_market_str<T: MatrixMarketScalar, S: MatrixMarketExport<T>>(
+    sparse_matrix: &S,
+) -> String {
+    let mut bytes = Vec::<u8>::new();
+    // This will call impl<A: Allocator> Write for Vec<u8, A>
+    // The vector will grow as needed.
+    // So, unwrap here won't cause any issue.
+    write_to_matrix_market(&mut bytes, sparse_matrix).unwrap();
```

Comment:
This comment makes me wonder if perhaps `write_to_matrix_market` shouldn't even return a `MatrixMarketError`, because the only thing that can go wrong are IO errors? Or is there something else? I would suppose that we should *always* be able to *export* matrix market data without errors, right?

Reply:
yes, I think the only thing could go wrong is IO error. Since I wrap up the error when loading the matrix by `load_from_file`, so I just thought to make it kind of consistent, also give an MatrixMarketError when `write_to_file`

Reply:
Maybe we should rename `MatrixMarketError` to `MatrixMarketParseError` and simply return only an IO error for export?